tests/functional/test_functional.py

```diff
@@ -80,19 +88,35 @@ def test_bot_upload_file_from_link(driver):
 
 def test_bot_reply_to_channel_message(driver):
     driver.send_channel_message('hello')
-    driver.wait_for_bot_channel_message('hello!')
+    driver.wait_for_bot_channel_message('hello sender!')
     driver.send_channel_message('hello', colon=False)
-    driver.wait_for_bot_channel_message('hello!')
+    driver.wait_for_bot_channel_message('hello sender!')
+
+def test_bot_listen_to_channel_message(driver):
+    driver.send_channel_message('hello', tobot=False)
+    driver.wait_for_bot_channel_message('hello channel!', tosender=False)
 
 def test_bot_reply_to_group_message(driver):
     driver.send_group_message('hello')
-    driver.wait_for_bot_group_message('hello!')
+    driver.wait_for_bot_group_message('hello sender!')
     driver.send_group_message('hello', colon=False)
-    driver.wait_for_bot_group_message('hello!')
+    driver.wait_for_bot_group_message('hello sender!')
 
-def test_bot_ignores_non_related_channel_message(driver):
+def test_bot_ignores_non_related_message_response_tosender(driver):
+    driver.send_channel_message('hello', tobot=True)
+    driver.ensure_only_specificmessage_from_bot('hello sender!', tosender=True)
+
+def test_bot_ignores_non_related_message_response_tochannel(driver):
     driver.send_channel_message('hello', tobot=False)
-    driver.ensure_no_channel_reply_from_bot()
+    driver.ensure_only_specificmessage_from_bot('hello channel!', tosender=False)
+
```

Comment:
Here we also need tests that checks "the bot must not send any reply if the channel message doesn't match any registered plugin". I think I have written that tests, but seems you have overwritten it?